### PR TITLE
[Dungeon] Gnomeregan

### PR DESCRIPTION
**Pull request by Knindzagxg**
Changes:

Min, Max & Modifier Damage for whole Dungeon.
Added flags for each boss, so they will appear with skull instead regular level elite.
Missing AI included now as well.

All checks have been done trough Classic Realms and Youtube videos.

How to test:
Apply SQL
Restart Server and delete your Cache.

Make sure to add very very basic gear towards each Dungeon.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1588337177136468800.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6232;
+
+/* RARE */
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 161, `maxdmg` = 207, `DamageModifier` = 1.02 WHERE `entry` = 6228;
+
+/* BOSS */
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 7361;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 90, `maxdmg` = 115, `DamageModifier` = 1.01 WHERE `entry` = 7079;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 6235;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 6229;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337177136468800.sql:38]:
```suggestion
UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (7361, 6235, 6229);
UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 90, `maxdmg` = 115, `DamageModifier` = 1.01 WHERE `entry` = 7079;
```

File: data/sql/updates/pending_db_world/rev_1588337177136468800.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+/*
+ * Dungeon: Gnomeregan
+ * Update by Knindza | <www.azerothcore.org>
+*/
+
+/* REGULAR */ 
+UPDATE `creature_template` SET `mindmg` = 56, `maxdmg` = 77, `DamageModifier` = 1.03 WHERE `entry` = 6329;
+UPDATE `creature_template` SET `mindmg` = 63, `maxdmg` = 92, `DamageModifier` = 1.03 WHERE `entry` = 6207;
+UPDATE `creature_template` SET `mindmg` = 63, `maxdmg` = 92, `DamageModifier` = 1.03 WHERE `entry` = 6206;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337177136468800.sql:11]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 63, `maxdmg` = 92, `DamageModifier` = 1.03 WHERE `entry` IN (6207, 6206);
```

File: data/sql/updates/pending_db_world/rev_1588337177136468800.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 7603;
+UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6222;
+UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6230;
+UPDATE `creature_template` SET `mindmg` = 46, `maxdmg` = 63, `minrangedmg` = 37, `maxrangedmg` = 58, `DamageModifier` = 1.03 WHERE `entry` = 6225;
+UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6224;
+UPDATE `creature_template` SET `mindmg` = 40, `maxdmg` = 54, `DamageModifier` = 1.03 WHERE `entry` = 6227;
+UPDATE `creature_template` SET `mindmg` = 47, `maxdmg` = 63, `DamageModifier` = 1.03 WHERE `entry` = 6226;
+UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6234;
+UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6212;
+UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6232;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337177136468800.sql:29]:
```suggestion
UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 92, `DamageModifier` = 1.03 WHERE `entry` IN (6211, 6218);
UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` IN (6219, 7603, 6222, 6224);
UPDATE `creature_template` SET `mindmg` = 58, `maxdmg` = 81, `DamageModifier` = 1.03 WHERE `entry` = 6220;
UPDATE `creature_template` SET `mindmg` = 65, `maxdmg` = 88, `DamageModifier` = 1.03 WHERE `entry` = 7849;
UPDATE `creature_template` SET `mindmg` = 65, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` = 6233;
UPDATE `creature_template` SET `mindmg` = 68, `maxdmg` = 95, `minrangedmg` = 38, `maxrangedmg` = 58, `DamageModifier` = 1.03 WHERE `entry` = 6223;
UPDATE `creature_template` SET `mindmg` = 71, `maxdmg` = 95, `DamageModifier` = 1.03 WHERE `entry` IN (6230, 6234, 6212, 6232);
UPDATE `creature_template` SET `mindmg` = 46, `maxdmg` = 63, `minrangedmg` = 37, `maxrangedmg` = 58, `DamageModifier` = 1.03 WHERE `entry` = 6225;
UPDATE `creature_template` SET `mindmg` = 40, `maxdmg` = 54, `DamageModifier` = 1.03 WHERE `entry` = 6227;
UPDATE `creature_template` SET `mindmg` = 47, `maxdmg` = 63, `DamageModifier` = 1.03 WHERE `entry` = 6226;
```

File: data/sql/updates/pending_db_world/rev_1588337177136468800.sql [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+UPDATE `creature_template` SET `mindmg` = 40, `maxdmg` = 54, `DamageModifier` = 1.03 WHERE `entry` = 6227;
+UPDATE `creature_template` SET `mindmg` = 47, `maxdmg` = 63, `DamageModifier` = 1.03 WHERE `entry` = 6226;
+
+/* RARE */
+UPDATE `creature_template` SET `rank` = 2, `mindmg` = 161, `maxdmg` = 207, `DamageModifier` = 1.02 WHERE `entry` = 6228;
+
+/* BOSS */
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (7361, 6235, 6229);
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 90, `maxdmg` = 115, `DamageModifier` = 1.01 WHERE `entry` = 7079;
+UPDATE `creature_template` SET `type_flags` = 4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 7800;
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1588337177136468800.sql:29]:
```suggestion
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` IN (7361, 6235, 6229);
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 90, `maxdmg` = 115, `DamageModifier` = 1.01 WHERE `entry` = 7079;
UPDATE `creature_template` SET `type_flags`=`type_flags`|4, `mindmg` = 193, `maxdmg` = 249, `DamageModifier` = 1.01 WHERE `entry` = 7800;
```

**Comment by Geekena** [comment]
and again i got lost
i forgot how big gnomeregan is big
damage and spells are ok except on tunel boss
again hes minion doing big damage

**Comment by Knindzagxg** [comment]
It's supposed to be once again mate.

Since boss it self don't have any mechanic or spell, Chomper will do some extra melee damage.

**Comment by Kitzunu** [comment]
Ty for the  PR!

**Comment by cts17** [comment]
The Walking Bomb at the final boss isn't doing damage. I don't know what the numbers should be, but definitely more than 0! Right now they just run up to you and then they just die.

**Pull request merged**